src/homework/sandro/FourInARow/FourInARow.java

```diff
@@ -0,0 +1,305 @@
+package homework.sandro.FourInARow;
+
+import homework.sandro.Colors;
+
+import java.util.Scanner;
+
+public class FourInARow {
+//    static {
+//        if (Math.random() > 0.5) {
+//            throw new RuntimeException("Hi Sandro, was geht ab?");
+//        }
+//    }
+
+    public static void main(String[] args) {
+
+        int[][] board = new int[6][7];
+        int currentplayer = 1;
+        boolean winner = false;
+        boolean empty = true;
+
+        printBoard(board, 6, 7);
+        while (!winner && empty) {                       //Game runs
+            int col;
+            int row;
+            while (true) {                               //Input read + check if legal move and column empty proof
+                col = readInt("pick a Col, Player Number: " + currentplayer);
+                row = emptyColumn(board, col);
+                if (row == -1) {
+                    System.out.println("try again, column is already full");
+                } else {
+                    break;
+                }
+            }
+
+
+            board[row][col] = currentplayer;                //update board with input and print new board
+            printBoard(board, 6, 7);
+
+
+            winner = hasWin(board, currentplayer);          //detect if currentplayer move trigger a win condition
+            empty = hasAnyMoreMoves(board, 6, 7);   //detect if board has empty spots, for Game is Draw detection
+
+            if (currentplayer == 1) {                       // player change, swap to player 2
+                currentplayer = 2;
+            } else {
+                currentplayer = 1;                          // player change, swap to player 1
+            }
+        }
+        if (winner) {                                       //if win condition is triggered, print
+            System.out.println("Winner" + currentplayer);
+        } else {                                            //if now win condition is triggered and no empty spaces left, print
+            System.out.println("Game is a Draw");
+        }
+    }
+
+
+    private static int emptyColumn(int[][] board, int col) {
+        for (int i = board.length - 1; i >= 0; i--) {
+            if (board[i][col] == 0) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    public static boolean hasWin(int[][] field, int currentplayer) {
+        for (int i = 0; i < field.length; i++) {
+            if (detectWinRow(field, i, currentplayer)
+                    || (detectWinCol(field, i, currentplayer)                     //what does "i" stand for (field, i, cur...)
```

Comment:
This is a bug. Your for loop goes from i = 0 to field.length. Field.length is equal to your row count = 6. But you have seven columns. => You never check the last column. When getting rid of the loop, you will get rid of that bug as well.